redis/test_test.go

```diff
@@ -54,7 +54,61 @@ type Server struct {
 	done chan struct{}
 }
 
+type Version struct {
+	major int
+	minor int
+	micro int
+}
+
+func GetRedisVersion() (*Version, error) {
+	out, err := exec.Command("redis-server", "--version").Output()
+	if err != nil {
+		return nil, err
+	}
+
+	version := ""
+	fields := strings.Fields(string(out))
+	for _, field := range fields {
+		if strings.HasPrefix(field, "v=") {
+			version = field[2:]
+			break
+		}
+	}
+	if version == "" {
+		return nil, errors.New("no version string")
+	}
+
+	values := [3]int{}
+	elems := strings.Split(version, ".")
+	for i, _ := range elems {
+		if i > 2 {
+			break
+		}
+		values[i], err = strconv.Atoi(elems[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	v := &Version{
+		major: values[0],
+		minor: values[1],
+		micro: values[2],
+	}
+
+	return v, nil
+}
```

Comment:
This can be simplified to the following. Notice a few things:
* It's idiomatic in go to avoid naming things `get` / `set`
* Wrap errors with `fmt.Errorf` to provide context to the user
```suggestion
func redisServerVersion() (*Version, error) {
	out, err := exec.Command("redis-server", "--version").Output()
	if err != nil {
		return nil, fmt.Errorf("server version: %w", err)
	}

	re := regexp.MustCompile(`v=(\d+)\.(\d+)\.(\d+)`)
	match := re.FindStringSubmatch(ver)
	if len(match) != 4 {
		return nil, fmt.Errorf("no server version found in %q", ver)
	}

	var err error
	var v version
	if v.major, err = strconv.Atoi(match[1]); err != nil {
		return nil, fmt.Errorf("invalid major version %q", match[1])
	}

	if v.minor, err = strconv.Atoi(match[1]); err != nil {
		return nil, fmt.Errorf("invalid minor version %q", match[2])
	}

	if v.patch, err = strconv.Atoi(match[1]); err != nil {
		return nil, fmt.Errorf("invalid patch version %q", match[3])
	}

	return &v, nil
}
```

Reply:
Little typo above, `strconv.Atoi(match[1])` repeated 3 times without incrementing the index.

Reply:
I don't know how to modify, so I'll commit the suggestion and edit it afterward.